# DIVA-MAGs
Distribution of variants across MAGs. Script for the Microbial Metagenomics laboratory, A.Y. 2021/2022.
This script is a tool for calculation of single nucleotide variation (SNV) density in metagenome assembled genomes (MAG), and is written in Python. It accepts MAGs in fasta format as input from InStrain, and provides graphical representations of the average of SNV in every position in each scaffold of each MAG.
This code was developed as part of a project carried out by Sofia Fraulini, Amin Zolfaghari, Carmelo Vittorio Musumarra and Faezeh Jalali during the course of Microbial Metagenomics (Molecular Biology Master degree) at the University of Padova. The project was supervised by Prof. Stefano Campanaro, Dr. Maria Silvia Morlino, Dr. Edoardo Bizzotto and Dr. Gabriele Ghiotto.
### Python dependencies:
 Please install these Python packages:
Pandas
Matplotlib
Numpy
Os
Argparse 
