# DIVA-MAGs
Distribution of variants across MAGs. Script for the Microbial Metagenomics laboratory, A.Y. 2021/2022.
This script is a tool for calculation of single nucleotide variation (SNV) density in metagenome assembled genomes (MAG), and is written in Python. It accepts MAGs in fasta format as input from [InStrain](https://github.com/MrOlm/inStrain), and provides graphical representations of the average of SNV in every position in each scaffold of each MAG.
This code was developed as part of a project carried out by Sofia Fraulini, Amin Zolfaghari, Carmelo Vittorio Musumarra and Faezeh Jalali during the course of Microbial Metagenomics (Molecular Biology Master degree) at the University of Padova. The project was supervised by Prof. Stefano Campanaro, Dr. Maria Silvia Morlino, Dr. Edoardo Bizzotto and Dr. Gabriele Ghiotto.
### Python dependencies:
 Please install these Python packages:
- Pandas
- Matplotlib
- Numpy
- Os
- Argparse 
### Usage:
In the first step for each MAG the following files are needed:
- _SNVs.tsv File generated by InStrain, containing genomic coordinates and information for each variant detected. Tab-separated.
- _scaffold_info.tsv File generated by InStrain, containing length and coverage metrics for each scaffold. Tab-separated.
- scaffold_to_bin.stb File generated by dRep, matching each scaffold to the corresponding MAG. Tab-separated.

Then make a text file. In the first line write the name of the folder that will be generated for saving the results, and write the path of each InStrain file with the above order.

In the next step pass the text file (e.g: FILE.txt) to the `--files` option like so:

`--files FILE.txt`

Only the above command is mandatory. Some of the following commands can be handy if you want to change a default value:

For help: 

`-h, --help`

Determine a threshold for coverage, by default the program uses the first quartile of the coverage distribution.

`--coverage_info, --c`

Define the size of window and step for each, if not given then the program manually creates window size and step for each MAG. Pay attention to this point that the size of the windows and steps will affect your final result and annotation.

` --window_info, --w`

Range of position to print in the final output, default is 20.

`  --printed_position`

Minimum number of variants needed to perform MAG analysis, otherwise will be discarded. Default is 20.

` --min_variants_MAG`

Minimum number of variants needed to perform scaffold analysis, otherwise discarded. Greater than 1, by default is 1.

`--min_variants_scaffold`

The script will provide a graph and a table for each MAG. In each graph the involved scaffolds are shown in different colors and separated by vertical lines.
This should produce a plot similar to the one below.

